### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by sovchenko**

### CODE REVIEW

File: src/main/java/com/softserveinc/atqc/page_elements/items_grid/ProductTile.java [hunk 1]
```diff
@@ -19,36 +20,52 @@
...
 
+    @Step("Add product to the wishlist")
     public WishListPage addProductToWishList() {
         addToWishListButton.click();
+
         return new WishListPage();
     }
 
+    @Step("Open product's reviews")
     public ProductReviewsTab openProductReviews() {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/atqc/page_elements/items_grid/ProductTile.java]: openProductReviewsTab()
fix step

File: src/main/java/com/softserveinc/atqc/page_elements/items_grid/ProductTile.java [hunk 2]
```diff
@@ -19,36 +20,52 @@
...
+    @Step("Add product to the comparison")
     public ProductComparisonPage addProductToComparison() {
         addToComparisonButton.click();
+
         return new ProductComparisonPage();
     }
 
     public int getReviewsAmount() {
-        return Integer.parseInt(productReviewsLink.getText().replaceAll("\\D", ""));
+        int reviewsAmount = Integer.parseInt(
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/atqc/page_elements/items_grid/ProductTile.java]: val
stat imp

File: src/main/java/com/softserveinc/atqc/page_elements/menu/MenuCategories.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package com.softserveinc.atqc.page_elements.menu;
+
+import lombok.Getter;
+
+@Getter
+public enum MenuCategories {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/atqc/page_elements/menu/MenuCategories.java]: singular

File: src/main/java/com/softserveinc/atqc/page_elements/menu/MenuCategories.java [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+    PhonesTVsElectronics("telefony-tv-i-ehlektronika"), ComputersLaptops("computers-notebooks");
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/atqc/page_elements/menu/MenuCategories.java]: code convention

File: src/main/java/com/softserveinc/atqc/page_elements/menu/MenuCategories.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    private String categoryName;
+
+    MenuCategories(String categoryName) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/atqc/page_elements/menu/MenuCategories.java]: @RequiredArgsConstructor

File: src/main/java/com/softserveinc/atqc/page_elements/menu/SubmenuCategories.java [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+@Getter
+public enum SubmenuCategories {
+    SmartPhones("smartfon"), MobilePhones("mob-phones");
+
+    private String subcategoryName;
+
+    SubmenuCategories(String subcategoryName) {
+        this.subcategoryName = subcategoryName;
+    }
+}
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/atqc/page_elements/menu/SubmenuCategories.java]: same as above

File: src/main/java/com/softserveinc/atqc/page_elements/product_filters/ManufacturerProductFilter.java [hunk 7]
```diff
@@ -1,49 +1,58 @@
...
 public class ManufacturerProductFilter {
 
     @Step("Select manufacturer")
     public ProductsListPage selectManufacturer(String manufacturer) {
-        SelenideElement manufacturerCheckbox = $x("//a[@class='checkbox-filter__link' and contains(@href,'producer="
-                + manufacturer.toLowerCase()
-                + "')]");
-        manufacturerCheckbox.click();
+
+        val checkboxXpath = String.format("//a[@class='checkbox-filter__link' and contains(@href,'producer=%s')]",
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/atqc/page_elements/product_filters/ManufacturerProductFilter.java]: stat imp

File: src/main/java/com/softserveinc/atqc/page_elements/product_filters/ManufacturerProductFilter.java [hunk 8]
```diff
@@ -1,49 +1,58 @@
...
 
+    @Step("Expand 'Manufacturer' filter section")
     public ManufacturerProductFilter expandManufacturerFilter() {
-        $x("//button[contains(text(),'Виробник') or contains(text(),'Производитель')]").click();
+        $x("//div[@class='sidebar-alphabet']/../../../button");
+
         return this;
     }
 
+    @Step("Hide manufacturer filter")
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/atqc/page_elements/product_filters/ManufacturerProductFilter.java:39]: fix steps, must be consistent

File: src/main/java/com/softserveinc/atqc/page_elements/product_filters/product_filter_values/Manufacturers.java [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+public enum Manufacturers {
+    Apple("Apple"), Samsung("Samsung"), Xiaomi("Xiaomi"), Huawei("Huawei"),
+    Meizu("Meizu"), LG("LG"), Asus("Asus");
+
+    private String name;
+
+    Manufacturers(String name) {
+        this.name = name;
+    }
+}
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/atqc/page_elements/product_filters/product_filter_values/Manufacturers.java]: same

Comment (toleksyn) [reply src/main/java/com/softserveinc/atqc/page_elements/product_filters/product_filter_values/Manufacturers.java]: use Enum.name()

File: src/main/java/com/softserveinc/atqc/page_objects/HomePage.java [hunk 10]
```diff
@@ -1,28 +1,32 @@
...
 import static com.codeborne.selenide.Selenide.*;
 
-@Getter
 public class HomePage extends BasePage {
 
     @Step("Open home page")
-    public HomePage openHomePage() {
-        open("https://rozetka.com.ua/");
+    public HomePage open() {
+        Selenide.open("https://rozetka.com.ua/");
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/atqc/page_objects/HomePage.java:17]: create a property file, move URL there

File: src/main/java/com/softserveinc/atqc/page_objects/HomePage.java [hunk 11]
```diff
@@ -1,28 +1,32 @@
...
+
         return new HomePage();
     }
 
-    @Step("Select product category in menu")
+    @Step("Hover on product category in menu")
     public HomePage hoverMenuCategory(String category) {
-        $x("//a[@class='menu-categories__link' and contains(@href,'" + category + "')]").hover();
+        String menuCategoryXpath = String.format("//a[@class='menu-categories__link' and contains(@href,'%s')]", category);
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/atqc/page_objects/HomePage.java]: val, stat imp

File: src/main/java/com/softserveinc/atqc/page_objects/HomePage.java [hunk 12]
```diff
@@ -1,28 +1,32 @@
...
+        $x(menuCategoryXpath).hover();
+
         return this;
     }
 
     @Step("Select product subcategory in menu")
     public ProductsListPage selectProductSubcategory(String subcategory) {
-        $x("//a[@class='menu__link' and contains(@href,'" + subcategory + "')]").click();
+        String subCategoryXpath = String.format("//a[@class='menu__link' and contains(@href,'%s')]", subcategory);
+        $x(subCategoryXpath).click();
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/atqc/page_objects/HomePage.java:35]: same

File: src/main/java/com/softserveinc/atqc/page_objects/ProductPage.java [hunk 13]
```diff
@@ -6,32 +6,38 @@
 
 import static com.codeborne.selenide.Selenide.$;
 import static com.codeborne.selenide.Selenide.$x;
+import static java.lang.Integer.parseInt;
 
-public class ProductPage {
+public class ProductPage extends BasePage {
     private SelenideElement reviewsTab = $x("//a[@class='product__rating-reviews']");
 
+    private int removeAllNonDigitsCharactersOfTheElementText(SelenideElement element) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/atqc/page_objects/ProductPage.java]: parseToText()

File: src/main/java/com/softserveinc/atqc/page_objects/ProductsListPage.java [hunk 14]
```diff
@@ -29,11 +31,20 @@
 
                     productTile.setAddToComparisonButton(tileElement.find(By.xpath(".//button[@class='compare-button']")));
                     productTile.setAddToWishListButton(tileElement.find(By.xpath(".//button[@class='wish-button']")));
-                    productTile.setProductAvailable(tileElement.find(By.cssSelector("div.goods-tile__availability")).getAttribute("class").contains("available"));
+
+                    productTile.setProductAvailable(tileElement.find(By.cssSelector("div.goods-tile__availability"))
+                            .getAttribute("class")
+                            .contains("available"));
+
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/atqc/page_objects/ProductsListPage.java]: split into vars

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/atqc/page_elements/product_filters/PriceRangeProductFilter.java [hunk 15]
```diff
@@ -1,32 +1,35 @@
...
+
 
 import static com.codeborne.selenide.Selenide.$x;
 
 public class PriceRangeProductFilter {
 
+    @Step("Hide price range filter")
     public PriceRangeProductFilter hidePriceRangeProductFilter() {
-        $x("//button[contains(text(),'Ціна') or contains(text(),'Цена')]").click();
+        $x("//ctg-filter-slider[@class='filter_layout_sidebar']/../../../../../button").click();
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/atqc/page_elements/product_filters/PriceRangeProductFilter.java]: avoid absolute locators

File: src/main/java/com/softserveinc/atqc/page_elements/product_page_elements/ProductBuyTogetherTab.java [hunk 16]
```diff
@@ -1,4 +1,5 @@
 package com.softserveinc.atqc.page_elements.product_page_elements;
 
 public class ProductBuyTogetherTab {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/atqc/page_elements/product_page_elements/ProductBuyTogetherTab.java]: this is not a page element

File: src/main/java/com/softserveinc/atqc/page_objects/UserAccountPage.java [hunk 17]
```diff
@@ -13,8 +13,10 @@
     private Header header = new Header();
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/atqc/page_objects/UserAccountPage.java]: you already have this from base page

File: src/test/java/com/softserveinc/atqc/tests/LoginTest.java [hunk 18]
```diff
@@ -20,14 +19,14 @@ public void verifySuccessfulLogIn() {
         val expectedUserName = "Тест";
         val loginLinkLabel = "увійдіть в особистий кабінет";
         val homePage = new HomePage()
-                .openHomePage()
+                .open()
                 .getHeader()
                 .logIn("ezbooksforme@gmail.com", "Pa55word");
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/atqc/tests/LoginTest.java]: move to properties

File: src/test/java/com/softserveinc/atqc/tests/LoginTest.java [hunk 19]
```diff
@@ -20,14 +19,14 @@ public void verifySuccessfulLogIn() {
...
         val nameOfLoggedInUser = homePage.getHeader().getLoggedInUserName();
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/atqc/tests/LoginTest.java]: revert Of

File: src/main/java/com/softserveinc/atqc/page_objects/ProductsListPage.java [hunk 20]
```diff
@@ -20,7 +21,8 @@
         productTileElements
                 .first(amount)
                 .forEach(tileElement -> {
-                    ProductTile productTile = new ProductTile();
+                    val productTile = new ProductTile();
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/atqc/page_objects/ProductsListPage.java]: pass tileElement here

File: src/main/java/com/softserveinc/atqc/page_objects/ProductsListPage.java [hunk 21]
```diff
@@ -29,11 +31,20 @@
...
+                    productTile.setProductAvailable(tileElement.find(By.cssSelector("div.goods-tile__availability"))
+                            .getAttribute("class")
+                            .contains("available"));
+
                     productTile.setProductLink(tileElement.find(By.xpath(".//span[@class='goods-tile__title']")));
-                    if (tileElement.find(By.xpath(".//div[@class='goods-tile__stars']")).exists()) {
-                        productTile.setProductRate(tileElement.find(By.xpath(".//div[@class='goods-tile__stars']")).find(By.cssSelector("svg")).getAttribute("aria-label"));
+
+                    val ratingXpath = ".//div[@class='goods-tile__stars']";
+                    if (tileElement.find(By.xpath(ratingXpath)).exists()) {
```
Comment (toleksyn) [inline src/main/java/com/softserveinc/atqc/page_objects/ProductsListPage.java]: stat imp
xpath() from Selenide

File: src/test/java/com/softserveinc/atqc/tests/ProductFilterTest.java [hunk 22]
```diff
@@ -21,13 +24,13 @@ public void setup() {
     @Test
     public void verifyThatFilteredItemHasCorrectPriceAndTitle() {
         val productTile = new HomePage()
-                .openHomePage()
-                .hoverMenuCategory("telefony")
-                .selectProductSubcategory("smartfon")
+                .open()
+                .hoverMenuCategory(PhonesTVsElectronics.getCategoryName())
```
Comment (toleksyn) [inline src/test/java/com/softserveinc/atqc/tests/ProductFilterTest.java]: hoverMenuCategory(PhonesTVsElectronics)

File: src/main/java/com/softserveinc/atqc/page_elements/items_grid/ProductTile.java (hunk 1 above)
Comment (sovchenko) [reply src/main/java/com/softserveinc/atqc/page_elements/items_grid/ProductTile.java]: done

File: src/main/java/com/softserveinc/atqc/page_elements/items_grid/ProductTile.java (hunk 2 above)
Comment (sovchenko) [reply src/main/java/com/softserveinc/atqc/page_elements/items_grid/ProductTile.java]: done

File: src/main/java/com/softserveinc/atqc/page_elements/menu/MenuCategories.java (hunk 3 above)
Comment (sovchenko) [reply src/main/java/com/softserveinc/atqc/page_elements/menu/MenuCategories.java]: done

File: src/main/java/com/softserveinc/atqc/page_elements/menu/MenuCategories.java (hunk 4 above)
Comment (sovchenko) [reply src/main/java/com/softserveinc/atqc/page_elements/menu/MenuCategories.java]: done

File: src/main/java/com/softserveinc/atqc/page_objects/HomePage.java (hunk 11 above)
Comment (sovchenko) [reply src/main/java/com/softserveinc/atqc/page_objects/HomePage.java]: done

File: src/main/java/com/softserveinc/atqc/page_objects/HomePage.java (hunk 12 above)
Comment (sovchenko) [reply src/main/java/com/softserveinc/atqc/page_objects/HomePage.java:35]: done

File: src/main/java/com/softserveinc/atqc/page_objects/ProductsListPage.java (hunk 14 above)
Comment (sovchenko) [reply src/main/java/com/softserveinc/atqc/page_objects/ProductsListPage.java]: done

File: src/main/java/com/softserveinc/atqc/page_elements/product_filters/PriceRangeProductFilter.java (hunk 15 above)
Comment (sovchenko) [reply src/main/java/com/softserveinc/atqc/page_elements/product_filters/PriceRangeProductFilter.java]: done

File: src/main/java/com/softserveinc/atqc/page_objects/UserAccountPage.java (hunk 17 above)
Comment (sovchenko) [reply src/main/java/com/softserveinc/atqc/page_objects/UserAccountPage.java]: done

File: src/main/java/com/softserveinc/atqc/page_elements/product_page_elements/ProductBuyTogetherTab.java (hunk 16 above)
Comment (sovchenko) [reply src/main/java/com/softserveinc/atqc/page_elements/product_page_elements/ProductBuyTogetherTab.java]: done

File: src/main/java/com/softserveinc/atqc/page_elements/menu/MenuCategories.java (hunk 5 above)
Comment (sovchenko) [reply src/main/java/com/softserveinc/atqc/page_elements/menu/MenuCategories.java]: done

File: src/main/java/com/softserveinc/atqc/page_objects/ProductPage.java (hunk 13 above)
Comment (sovchenko) [reply src/main/java/com/softserveinc/atqc/page_objects/ProductPage.java]: done

File: src/main/java/com/softserveinc/atqc/page_objects/ProductsListPage.java (hunk 21 above)
Comment (sovchenko) [reply src/main/java/com/softserveinc/atqc/page_objects/ProductsListPage.java]: done

File: src/test/java/com/softserveinc/atqc/tests/ProductFilterTest.java (hunk 22 above)
Comment (sovchenko) [reply src/test/java/com/softserveinc/atqc/tests/ProductFilterTest.java]: done

**Pull request closed without merging**